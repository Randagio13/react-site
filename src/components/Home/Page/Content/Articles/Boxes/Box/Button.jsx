import styled from 'styled-components'

const Button = styled.button`
  border: 1px solid #3A405B;
  border-radius: 8px;
  font-family: Helvetica;
  font-size: 14px;
  color: #000000;
  letter-spacing: 2px;
  text-align: center;
  line-height: 16px;
  background: transparent;
  padding: 20px 79px 20px 79px;
  text-transform: uppercase;
  position: relative;
  top: -35px;
`
export default Button
