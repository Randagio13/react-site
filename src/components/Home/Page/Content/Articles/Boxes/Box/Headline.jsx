import styled from 'styled-components'

const Headline = styled.h1`
  font-family: Helvetica;
  font-size: 14px;
  color: #000000;
  letter-spacing: 0;
  line-height: 16px;
  text-align: left;
  width: 80%;
  align-self: center;
  padding: 20px 0 0 0;
  margin: 0;
  z-index: 1;
`
export default Headline
