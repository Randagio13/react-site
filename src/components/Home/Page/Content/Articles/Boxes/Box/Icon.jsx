import styled from 'styled-components'
import piggy from '../../../../../../../images/piggy.svg'

const Icon = styled.img.attrs(({ typeIcon }) => ({
  src: typeIcon === 'piggy' ? piggy : null
}))`
  align-self: flex-start;
  padding: 20px 0 0 30px;
  z-index: 1;
`

export default Icon
