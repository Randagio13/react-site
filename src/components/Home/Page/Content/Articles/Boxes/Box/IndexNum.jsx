import styled from 'styled-components'

const Index = styled.span`
  &::before {
    content: '${props => props.index}';
  }
  transform: translate(-60px, 40px);
  font-family: Helvetica;
  font-size: 216px;
  color: #ECDA1F;
  letter-spacing: 0;
  z-index: 0;
`
export default Index
