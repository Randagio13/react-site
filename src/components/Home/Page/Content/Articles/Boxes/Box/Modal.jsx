import styled from 'styled-components'

const ModalContainer = styled.div`
  transition: all 0.7s;
`

export default ModalContainer
