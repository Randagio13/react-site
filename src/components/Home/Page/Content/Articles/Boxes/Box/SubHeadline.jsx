import styled from 'styled-components'

const SubHeadline = styled.h3`
  font-family: Helvetica;
  font-size: 26px;
  color: #000000;
  letter-spacing: 0;
  line-height: 32px;
  text-align: left;
  width: 80%;
  align-self: center;
  margin: 0;
  padding: 15px 0 0 0;
  z-index: 1;
`
export default SubHeadline
