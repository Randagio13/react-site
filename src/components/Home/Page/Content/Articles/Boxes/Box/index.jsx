import styled, { css } from 'styled-components'
import Index from './IndexNum'
import Icon from './Icon'
import React, { Component } from 'react'
import Headline from './Headline'
import SubHeadline from './SubHeadline'
import Button from './Button'
import ModalContainer from './Modal'
import PropTypes from 'prop-types'

const openModal = css`
  transform: scale(1.5) translate(-10vw, -10vh);
`

const BoxStyled = styled.div`
  width: 310px;
  height: 414px;
  // width: ${props => props.isModal ? '957px' : '310px'};
  // height: ${props => props.isModal ? '658px' : '414px'};
  // ${props => props.isModal ? 'top: -30%' : 'top: 442px'};
  // ${props => props.isModal ? 'left: -200%' : 'left: 450px'};
  // ${props => props.isModal ? 'position: absolute' : 'position: fixed'};
  // ${props => props.isModal ? 'z-index: 1;' : ''}
  ${props => props.isModal ? `${openModal}` : ''}
  background: #F8E627;
  border-radius: 12px;
  font-family: Helvetica;
  font-size: 216px;
  color: #ECDA1F;
  letter-spacing: 0;
  font-family: Helvetica;
  font-size: 26px;
  color: #000000;
  letter-spacing: 0;
  line-height: 32px;
  margin: 0 0 30px 0;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-line-clamp: 2; /* number of lines to show */
  -webkit-box-orient: vertical;
  transition: all 0.5s;
`

class Box extends Component {
  constructor (props) {
    super(props)
    this.state = {
      isModal: false
    }
  }
  render () {
    const { indexNum, title, subTitle, icon, btnLabel } = this.props
    const { isModal } = this.state
    const excerpt = subTitle ? subTitle.substring(0, 70) : ''
    return (
      <BoxStyled isModal={isModal}>
        <Index>{indexNum}</Index>
        <Icon typeIcon={icon} />
        <Headline>{title}</Headline>
        <SubHeadline>{excerpt}</SubHeadline>
        <Button>{btnLabel}</Button>
        <ModalContainer>ciao</ModalContainer>
      </BoxStyled>
    )
  }
}

Box.propTypes = {
  /** Background number. */
  indexNum: PropTypes.string.isRequired,
  /** Title component. */
  title: PropTypes.string.isRequired,
  /** Subtitle component. */
  subTitle: PropTypes.string,
  /** Icon component. */
  icon: PropTypes.string,
  /** Icon component. */
  btnLabel: PropTypes.string,
  /** Show button component. */
  showButton: PropTypes.bool
}

Box.defaultProps = {
  btnLabel: 'clicca qui',
  subTitle: ''
}

export default Box
