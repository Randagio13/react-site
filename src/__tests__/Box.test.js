import React from 'react'
import ReactDOM from 'react-dom'
import Box from '../components/Home/Page/Content/Articles/Boxes/Box'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <Box indexNum='01' title='Test title' subTitle='Test sub' icon='puggy' />
    , div)
  ReactDOM.unmountComponentAtNode(div)
})

it('Open modal', () => {
  const comp = renderer.create(
    <Box indexNum='01' title='Test title' subTitle='Test sub' icon='puggy' />
  )
  let tree = comp.toJSON()
  expect(tree).toMatchSnapshot()
})
