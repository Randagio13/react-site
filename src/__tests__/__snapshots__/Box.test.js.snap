// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Open modal 1`] = `
<div
  className="sc-bZQynM cDchzc"
>
  <span
    className="sc-bdVaJa hUrPbB"
  >
    01
  </span>
  <img
    className="sc-bwzfXH kbkKXG"
    src={null}
  />
  <h1
    className="sc-htpNat fkEake"
  >
    Test title
  </h1>
  <h3
    className="sc-bxivhb xNlgE"
  >
    Test sub
  </h3>
  <button
    className="sc-ifAKCX glzoNA"
  >
    clicca qui
  </button>
  <div
    className="sc-EHOje jEOkGS"
  >
    ciao
  </div>
</div>
`;
